js-rattler/crate/solve.rs

```diff
@@ -61,20 +81,107 @@ pub async fn simple_solve(
+        })
+        .collect();
+
+    let installed_packages: Option<Vec<RepoDataRecord>> =
+        if locked_packages.is_null() || locked_packages.is_undefined() {
+            None
+        } else {
+            let js_locked_packages: Vec<JsLockedPackage> =
+                serde_wasm_bindgen::from_value(locked_packages).map_err(JsError::from)?;
+
+            let records = js_locked_packages
+                .into_iter()
+                .map(|pkg| {
+                    let url = Url::parse(&pkg.url)
+                        .map_err(|e| JsError::from(ParseChannelError::from(e)))?;
+
+                    let rec = PackageRecord {
+                        name: PackageName::try_from(pkg.package_name.clone())?,
+                        version: Version::from_str(&pkg.version)?.into(),
+                        build: pkg.build.clone(),
+                        build_number: pkg.build_number,
+                        subdir: "unknown".to_string(),
+                        md5: None,
+                        sha256: None,
+                        size: None,
+                        arch: None,
+                        platform: None,
+                        depends: pkg.depends.unwrap_or_default(),
+                        extra_depends: BTreeMap::new(),
+                        constrains: vec![],
+                        track_features: vec![],
+                        features: None,
+                        noarch: NoArchType::none(),
+                        license: None,
+                        license_family: None,
+                        timestamp: None,
+                        python_site_packages_path: None,
+                        legacy_bz2_md5: None,
+                        legacy_bz2_size: None,
+                        purls: None,
+                        run_exports: None,
+                    };
+
+                    Ok(RepoDataRecord {
+                        url,
+                        file_name: pkg.filename,
+                        channel: pkg.repo_name,
+                        package_record: rec.clone(),
+                    })
+                })
+                .collect::<Result<Vec<_>, JsError>>()?;
+
+            Some(records)
+        };
+
+    // if a locked package does not include depends then depends will be taken from repodata
+    let enriched_installed_packages: Vec<RepoDataRecord> = installed_packages
+        .unwrap_or_default()
+        .into_iter()
+        .map(|mut rec| {
+            let key = (
+                rec.package_record.name.as_normalized().to_string(),
+                rec.package_record.version.to_string(),
+                rec.package_record.build.clone(),
+            );
+
+            if rec.package_record.depends.is_empty() {
+                if let Some(deps) = repodata_keys.get(&key) {
+                    rec.package_record.depends = deps.clone();
+                }
+            }
+
+            rec
+        })
+        .collect();
```

Comment:
You are recollecting the data instead of just modifying the existing record. You could instead use `iter_mut`